lumispy/signals/luminescence_spectrum.py

```diff
@@ -344,6 +345,45 @@ def remove_background_from_file(self, background=None, inplace=False, **kwargs):
                 return self.map(lambda s, bkg: s - bkg, bkg=bkg_y, inplace=True)
 
 
+    SAVETXT_EXAMPLE = \
+    """
+    Examples
+    --------
+    >>> import lumispy as lum
+    >>> import numpy as np
+    
+    # Spectrum:
+    >>> s = lum.signals.LumiSpectrum(np.arange(5))
+    >>> s.savetxt('spectrum.txt')
+    # 0.00000	0.00000
+    # 1.00000	1.00000
+    # 2.00000	2.00000
+    # 3.00000	3.00000
+    # 4.00000	4.00000
+    
+    # Linescan:
+    >>> l = lum.signals.LumiSpectrum(np.arange(25).reshape((5,5)))
+    >>> l.savetxt('linescan.txt')
+    # 0.00000	0.00000	1.00000	2.00000	3.00000	4.00000
+    # 0.00000	0.00000	5.00000	10.00000	15.00000	20.00000
+    # 1.00000	1.00000	6.00000	11.00000	16.00000	21.00000
+    # 2.00000	2.00000	7.00000	12.00000	17.00000	22.00000
+    # 3.00000	3.00000	8.00000	13.00000	18.00000	23.00000
+    # 4.00000	4.00000	9.00000	14.00000	19.00000	24.00000
+    """
+
+    def savetxt(self, filename, fmt='%.5f', delimiter='\t', axes=True,
```

Comment:
That's a personal thought but worth mentioning...
I like to keep my data in pandas dataframes many times as I love using seaborn for plotting trends.

Could you add a parameter (if you think it's not too confusing) called `return_pandas_df` which returns the Pandas dataframe object  instead of saving a txt file (the same as you would save as txt).
That would save me to have to run two lines of code like that:

```
cl.savetxt('file.txt')
df = pd.read_csv('file.txt')
```

Reply:
sure, I can do so - then we can take pandas conventions as a guideline for the third dimension.

Reply:
I would maybe call it `return_array` and then you would have to run the line 
```
df = pd.DataFrame(cl.savetxt(return_array=True))
```

This avoids introducing pandas as dependency, but pandas could be mentioned as use case in the documentation.

Though it is a bit weird to run a `savetxt` function for that. Alternatively, I could put that part into a `to_array` function (open for better name proposals) with the same parameters that is called by `savetxt` but can also be called separately for the pandas use case. What do you think?

Reply:
I agree that we may be mixing functions... Let's forget about pandas for now I guess.
Maybe if you managed to split the save_txt function as a `to_array` + `savetxt` functions, then we could just call the first half of the workflow?
Does that make sense?

Reply:
That is what I thought, see update